postUserAuthentication/checkIPWithAbuseIPDBWorkflow.ts

```
@@ -0,0 +1,396 @@
+        fetchedData = response;
+      }
+
+      // Handle server errors gracefully
+      if (fetchedData && fetchedData.message === 'Server Error') {
+        if (config.failOpen) {
+          console.warn('AbuseIPDB API server error. Fail-open mode enabled - allowing access.');
+          abuseConfidenceScore = 0;
+        } else {
+          throw new Error('AbuseIPDB API server error. Access blocked.');
+        }
+      } else if (fetchedData && fetchedData.errors && fetchedData.errors.length > 0) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)